dataflowAPI/src/liveness.C

```diff
@@ -623,10 +623,13 @@ void LivenessAnalyzer::clean(Function *func){
 }
 
 bool LivenessAnalyzer::isMMX(MachRegister machReg){
-	if ((machReg.val() & Arch_x86) == Arch_x86 || (machReg.val() & Arch_x86_64) == Arch_x86_64){
-		assert( ((machReg.val() & x86::MMX) == x86::MMX) == ((machReg.val() & x86_64::MMX) == x86_64::MMX) );
+	auto const arch = machReg.getArchitecture();
+	if (arch == Arch_x86) {
 		return (machReg.val() & x86::MMX) == x86::MMX;
```

Comment:
The old test is not correct.  MMX has the value `0x00090000` so any category with those two bits set matches so second bytes of `*9` `*B` `*D` and `*F` matches.  The correct expression should be `(machReg.val() & 0x00ff0000) == x86::MMX`.

Reply:
Oh my. I thought that check looked weird, but I was just heading in to fix the crazy assert. Good catch.